src/io/flutter/preview/PreviewArea.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.preview;
+
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.actionSystem.ex.CustomComponentAction;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.SimpleToolWindowPanel;
+import com.intellij.ui.components.JBLabel;
+import com.intellij.util.ui.JBUI;
+import com.intellij.util.ui.UIUtil;
+import io.flutter.dart.FlutterDartAnalysisServer;
+import io.flutter.editor.*;
+import io.flutter.inspector.InspectorService;
+import io.flutter.inspector.InspectorGroupManagerService;
+import org.dartlang.analysis.server.protocol.FlutterOutline;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Class that manages displaying the DeviceMirror.
+ */
+public class PreviewArea {
```

Comment:
Are DeviceMirror and PreviewArea synonymous? If so, consider renaming this class and related fields to DeviceMirror for consistency

Reply:
I've filed a bug to rename these and related classes.
https://github.com/flutter/flutter-intellij/issues/4162

The existing "PreviewView" for example is really the OutlineView and everywhere else we say preview we should really say DeviceMirror.